internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -177,17 +191,20 @@ func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkf
 	dwctx, done := workflow.NewDisconnectedContext(wctx)
 	defer done()
 
-	sessionDurationHistogram.Record(
-		metricsCtx,
-		duration.Milliseconds(),
-		metric.WithAttributes(
-			attribute.Bool("replay", isReplaying),
-			attribute.Bool("success", err == nil),
-		),
-	)
+	sessionDurationHistogram.Record(metricsCtx, duration.Milliseconds(),
+		metric.WithAttributes(attribute.Bool("replay", isReplaying), attribute.Bool("success", err == nil)))
 
 	l = l.With(zap.Duration("duration", duration))
 
+	if eventTime != (time.Time{}) {
```

Comment:
what does this check ?
event time should always be set

Reply:
this is the check we discussed yesterday. I may fail to find event time in inputs.. so if I failed to check then no metric will be added as you requested.

Reply:
Checking the result of a code block 60 lines above here is not acceptable. No one should be expected to understand that a value check in line 200 depends on code in line 140.

If you **have** to, you can use a self-documenting bool (e.g. `createTimeFound`) - it's not ideal either, but at least it explains the logical flow.

Reply:
I could use a bool if coz. But you said this is not ideal.. Could u please suggest the ideal solution for this case?